packages/cli-lib/lib/uploadFolder.js

```diff
@@ -28,6 +31,91 @@ const queue = new PQueue({
   concurrency: 10,
 });
 
+function getFilesByTypeAndProcessFields(files, src, writeDir = src) {
+  const writeDirRegex = new RegExp(`^${escapeRegExp(src)}`);
+  const moduleFiles = [];
+  const cssAndJsFiles = [];
+  const otherFiles = [];
+  const templateFiles = [];
+  const jsonFiles = [];
+  const compiledJsonFiles = [];
+
+  const fieldsJsInRoot = listFilesInDir(src).includes('fields.js');
+  const options = yargs.argv.options;
+
+  files.forEach(file => {
+    const parts = splitLocalPath(file);
+    const extension = getExt(file);
+    const moduleFolder = parts.find(part => part.endsWith('.module'));
+    const fileName = parts[parts.length - 1];
+    const relativePath = file.replace(writeDirRegex, '');
+
+    if (fileName == 'fields.output.json') {
```

Comment:
With the move to `tmp` dir, do we need this?

Reply:
Yes, if the user has `--saveOutput` (which is enabled by default for now) then we do write the `fields.output.json` to the same folder as the `fields.js`

Reply:
Ahhhh right, sorry I forgot about that.